frontend/nexusInteraction.cpp

```diff
@@ -395,4 +427,21 @@ void MainWindow::handle_result_contextGraph(HttpRequestWorker *worker)
     }
 }
 
-void MainWindow::removeTab(int index) { delete ui_->tabWidget->widget(index); }
\ No newline at end of file
+void MainWindow::removeTab(int index) { delete ui_->tabWidget->widget(index); }
+
+void MainWindow::toggleAxis(int state)
+{
+    auto *toggleBox = qobject_cast<QCheckBox *>(sender());
+    auto *graphParent = toggleBox->parentWidget();
+    QList<QChartView *> help = graphParent->findChildren<QChartView *>();
+    if (toggleBox->isChecked())
+    {
+        help[0]->chart()->axes()[2]->hide();
+        help[0]->chart()->axes()[3]->show();
+    }
+    else
+    {
+        help[0]->chart()->axes()[2]->show();
+        help[0]->chart()->axes()[3]->hide();
+    }
+}
```

Comment:
This would be further neatened if you create your own compound widget containing the `QChartView` and any associated controls (like the axis toggle).